### PR TITLE
Update solc v0.6

### PR DESCRIPTION
**Pull request by nlordell**
Update the contracts to the latest solc. This is possible with the release of openzeppelin 3.0

### Test Plan

CI - no significant code changes.

### CODE REVIEW

**Review by e00E — APPROVED** [review]

**Comment by nlordell** [comment]
@e00E do you mind taking a quick look at the hack I had to add here:  https://github.com/gnosis/ethcontract-rs/pull/282/files#diff-0f582719fa997c49e0b2aab86afe0390R107-R135

File: common/src/truffle.rs [hunk 1]
```diff
@@ -103,3 +104,32 @@ mod tests {
         }
     }
 }
+
+/// Deserialization implementation for compatibility with ABIs produced with
+/// solc v0.6. This is a known `ethabi` issue with an open PR for a fix:
+/// https://github.com/openethereum/ethabi/issues/185
+/// https://github.com/openethereum/ethabi/pull/187
```
Comment (e00E) [inline common/src/truffle.rs:111]: Should we wait for ethabi to update with the fix or is it important to update solc quickly?

File: common/src/truffle.rs [hunk 2]
```diff
@@ -103,3 +104,32 @@ mod tests {
...
+    where
+        D: Deserializer<'de>,
+    {
+        let mut raw_abi: Vec<Value> = Deserialize::deserialize(deserializer)?;
+        let receive = json!("receive");
+        for entry in raw_abi.iter_mut() {
+            if let Value::Object(obj) = entry {
+                if let Some(kind) = obj.get_mut("type") {
+                    if kind == &receive {
+                        *kind = json!("fallback");
```
Comment (e00E) [inline common/src/truffle.rs:127]: Is this correct when there already is a fallback function?

Comment (nlordell) [reply common/src/truffle.rs:127]: `receive` is a payable fallback function from what I understand. Currently `ethabi` doesn't differentiate between payable and non-payable fallback functions and allows more than one in the ABI, so this should be fine.

File: common/src/truffle.rs (hunk 1 above)
Comment (nlordell) [reply common/src/truffle.rs:111]: Good point, but that would mean that `ethcontract` doesn't support v0.6.0 compiled contracts. The PR has been open for a couple weeks without any discussion so I would prefer having this and just removing it once it gets merged.

File: common/src/truffle.rs (hunk 2 above)
Comment (e00E) [reply common/src/truffle.rs:127]: If it allows more than one then it seems fine.

**Review by e00E — APPROVED** [review]

File: examples/truffle/contracts/RustCoin.sol [hunk 3]
```diff
@@ -1,14 +1,13 @@
-pragma solidity ^0.5.0;
+pragma solidity ^0.6.0;
 
 import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
-import "@openzeppelin/contracts/token/ERC20/ERC20Detailed.sol";
 
-contract RustCoin is ERC20, ERC20Detailed {
-  constructor() ERC20Detailed("Rust Coin", "RUST", 18) public {
+contract RustCoin is ERC20 {
+  constructor() ERC20("Rust Coin", "RUST") public {
```
Comment (bh2smith) [inline examples/truffle/contracts/RustCoin.sol:6]: I see you noticed that Detailed was merged with ERC20. Nice catch. How did you find out? But wait, isn't this an open-zeppelin specific new thing?

File: examples/truffle/contracts/RustCoin.sol [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
     _mint(msg.sender, 1337 * (10 ** uint256(decimals())));
   }
 
-  function() external payable {
+  receive() external payable {
```
Comment (bh2smith) [inline examples/truffle/contracts/RustCoin.sol:10]: This must also be some sort of fancy new feature.

File: examples/truffle/package.json [hunk 5]
```diff
@@ -23,7 +23,7 @@
   "homepage": "https://github.com/gnosis/ethcontract-rs",
   "devDependencies": {
     "@gnosis.pm/util-contracts": "^2.0.6",
-    "@openzeppelin/contracts": "^2.5.0",
+    "@openzeppelin/contracts": "^3.0.0",
```
Comment (bh2smith) [inline examples/truffle/package.json:26]: Oh. I see you already saw that too! Are you a wizard?

**Review by bh2smith — APPROVED** [review]
Staying very on top of things around here. Gods gift to open source.

File: examples/truffle/package.json (hunk 5 above)
Comment (nlordell) [reply examples/truffle/package.json:26]: dependabot told me :wink:

File: examples/truffle/contracts/RustCoin.sol (hunk 4 above)
Comment (nlordell) [reply examples/truffle/contracts/RustCoin.sol:10]: `receive` is a payable fallback function. So before you had:
```
function() external {
    // non-payable fallback function
}
function() external payable {
    // payable fallback function
}
```
and now you have
```
fallback() external {
    // non-payable fallback function
}
receive() external payable {
    // payable fallback function
}
```

File: examples/truffle/contracts/RustCoin.sol (hunk 3 above)
Comment (nlordell) [reply examples/truffle/contracts/RustCoin.sol:6]: Yes, we use the openzeppeling ERC20 implementation for one of our samples.

**Comment by nlordell** [comment]
@e00E I'm going to leave this open and on-hold to see if an `ethabi` update comes in that fixes the `receive` function support.

**Pull request merged**